app/Http/Middleware/RedirectIfAuthenticated.php

```diff
@@ -9,16 +9,16 @@ class RedirectIfAuthenticated {
     /**
      * Handle an incoming request.
      *
-     * @param  \Illuminate\Http\Request  $request
-     * @param  \Closure  $next
-     * @param  string|null  $guard
-     * @return mixed
+     * @param  \Closure(\Illuminate\Http\Request): (\Symfony\Component\HttpFoundation\Response)  $next
      */
-    public function handle($request, Closure $next, $guard = null) {
-        if (Auth::guard($guard)->check()) {
-            return redirect()->back();
-        }
+    public function handle(Request $request, Closure $next, string ...$guards): Response {
+        $guards = empty($guards) ? [null] : $guards;
```

Comment:
I don't think this should be necessary because if `$guards` is empty, then shouldn't the foreach just be skipped?

Reply:
This code block is straight from the Laravel 10.x repo... I'm not sure what protection this provides, but someone apparently someone thinks that $guards could be a non-null empty type of some kind. https://github.com/laravel/laravel/blob/10.x/app/Http/Middleware/RedirectIfAuthenticated.php

Reply:
Hmm... I'd like to see if we can get away without that line. But if it breaks something and we need it, I'm not super worried about it

Reply:
I'd prefer that we leave this here. It is part of the 10.x repo and it just provides extra protection from unexpected variable input. I don't fully understand the implications, so I'd prefer that it remain since someone on the Laravel team agreed to include it.